src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function add(number) {
+    if (typeof number === 'number') {
+      sum += number;
+    } else {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+
+    return add;
+  }
+
+  return add;
```

Comment:
The `add` function should not check the type of `number`. According to the task description, there is no need to verify that the argument is a number, so the type check in line 10 can be removed. Instead, you should check if `number` is `undefined` to know when to return the result and reset the sum.